globus_action_provider_tools/authentication.py

```diff
@@ -177,15 +171,20 @@ def get_authorizer_for_scope(
         except KeyError:
             return None
 
-        if "refresh_token" in dep_tkn_resp:
+        refresh_token = dep_tkn_resp.get("refresh_token")
+        access_token = dep_tkn_resp.get("access_token")
+        token_expiration = dep_tkn_resp.get("expires_at_seconds")
+        now = time()
+
+        if refresh_token is not None:
             return RefreshTokenAuthorizer(
-                dep_tkn_resp["refresh_token"],
+                refresh_token,
                 self.auth_client,
-                access_token=dep_tkn_resp["access_token"],
-                expires_at=dep_tkn_resp["expires_at_seconds"],
+                access_token=access_token,
+                expires_at=token_expiration,
             )
-        elif "access_token" in dep_tkn_resp:
-            return AccessTokenAuthorizer(dep_tkn_resp["access_token"])
+        elif access_token is not None and token_expiration > now:
+            return AccessTokenAuthorizer(access_token)
```

Comment:
In practice it seems unlikely that you'll have a response that contains an `"access_token"` key but no `"token_expiration"` key. Nevertheless, this is not verifying the type of `token_expiration` and may raise a `TypeError` as a result.

Reply:
Good one, I ended up putting some semi-convoluted conversion to int attempt